cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -2868,8 +2886,19 @@ class AzureFileSystem::Impl {
     if (src == dest) {
       return Status::OK();
     }
+    std::string sas_token;
+    {
+      Storage::Sas::BlobSasBuilder builder;
+      std::chrono::seconds available_period(60);
+      builder.ExpiresOn = std::chrono::system_clock::now() + available_period;
+      builder.BlobContainerName = src.container;
+      builder.BlobName = src.path;
+      builder.Resource = Storage::Sas::BlobSasResource::Blob;
+      builder.SetPermissions(Storage::Sas::BlobSasPermissions::Read);
+      ARROW_ASSIGN_OR_RAISE(sas_token, options_.GenerateSASToken(&builder));
+    }
+    auto src_url = GetBlobClient(src.container, src.path).GetUrl() + sas_token;
```

Comment:
What is a sas token that it can be concatenated just like this to an URL?

Reply:
A sample SAS token: `?se=2024-04-18T21:17:19Z&sig=HI/8c...ZExA%3D&sp=r&spr=https&sr=b&sv=2022-11-02`

Reply:
Ok. It comes with the `?` already.